uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Upgrader/Steps/1032.php

```diff
@@ -0,0 +1,164 @@
+      'label' => 'Add New Staff Member',
+    ]);
+
+    $menu = array_shift($menu['values']);
+    civicrm_api3('Navigation', 'create', [
+      'label' => 'Staff Directory',
+      'name' => 'Staff Directory',
+      'url' => 'civicrm/contact/search/advanced?reset=1&force=1',
+      'parent_id' => $menu['parent_id'],
+      'domain_id' => $menu['domain_id'],
+      'permission' => $menu['permission'],
+      'is_active' => 1,
+      'weight' => 2,
+      'has_separator' => 1,
+    ]);
+    // If we don't flush it will not recognize newly created parent_id
+    CRM_Core_PseudoConstant::flush();
+  }
+
+  /**
+   * Creates a new menu heading for Communications
+   */
+  private function up1032_createCommunicationsMenu() {
+    $menu = civicrm_api3('Navigation', 'get', [
+      'label' => 'Add New Staff Member',
+    ]);
+
+    $menu = array_shift($menu['values']);
+    civicrm_api3('Navigation', 'create', [
+      'label' => 'Record Communication',
+      'name' => 'Record Communication',
+      'parent_id' => $menu['parent_id'],
+      'domain_id' => $menu['domain_id'],
+      'permission' => $menu['permission'],
+      'is_active' => 1,
+      'weight' => 3,
+    ]);
+    // If we don't flush it will not recognize newly created parent_id
+    CRM_Core_PseudoConstant::flush();
+  }
+
+  /**
+   * Moves the New Email submenu to be child of Communications
+   */
+  private function up1032_moveNewEmailSubmenu() {
+    $menu = civicrm_api3('Navigation', 'get', [
+      'label' => 'Record Communication',
+    ]);
+
+    $menu = array_shift($menu['values']);
+    civicrm_api3('Navigation', 'get', [
+      'label' => 'New Email',
+      'api.Navigation.create' => [
+        'id' => '$value.id',
+        'parent_id' => $menu['id'],
+        'has_separator' => 0,
+        'weight' => 1,
+      ],
+    ]);
+  }
+
+  /**
+   * Creates a new submenu item under Communications
+   */
+  private function up1032_createCommunicationsSubmenus() {
+    $menu = civicrm_api3('Navigation', 'get', [
+      'label' => 'Record Communication',
+    ]);
+
+    $menu = array_shift($menu['values']);
+    civicrm_api3('Navigation', 'create', [
+      'label' => 'New Meeting',
+      'name' => 'New Meeting',
+      'parent_id' => $menu['id'],
+      'domain_id' => $menu['domain_id'],
+      'permission' => $menu['permission'],
+      'is_active' => 1,
+      'weight' => 2,
+      'url' => '/activity?reset=1&action=add&context=standalone',
+    ]);
+    // If we don't flush it will not recognize newly created parent_id
+    CRM_Core_PseudoConstant::flush();
+  }
+
+  /**
+   * Removes the old menu items under Staff Menu
+   *
+   * @param array $menus
+   */
+  private function up1032_removeOldMenuItems($menus) {
+    foreach ($menus as $menuLabel) {
+      civicrm_api3('Navigation', 'get', [
+        'label' => $menuLabel,
+        'api.Navigation.delete' => ['id' => '$value.id'],
+      ]);
+    }
+  }
+
+  /**
+   * Disables the Import Menu items. They will be moved later
```

Comment:
I don't think `They will moved later` should be part of what the function does.
You can add as comment in the code snippet. 🤷‍♂️